src/ctelnet.cpp

```diff
@@ -2031,6 +2066,15 @@ void cTelnet::processTelnetCommand(const std::string& telnetCommand)
             sendTelnetOption(TN_WILL, OPT_MSDP);
             raiseProtocolEvent("sysProtocolEnabled", "MSDP");
             break;
+        } else if (option == OPT_MSDP) {
+            sendTelnetOption(TN_WONT, OPT_MSDP);
```

Comment:
Would it not be clearer if the `mpHost->mEnableMSDP` was in a separate, nested `if ()` inside an `if (option == OPT_MSDP)`?

Reply:
And the same for the cases that follow for other options...

Reply:
The if blocks have been refactored into a cleaner, single-condition form for each Telnet option (MSDP, ATCP, GMCP, MSSP, MSP, and MXP) without changing the logic. This improves readability while preserving exact behavior. Let me know if you’d like similar simplifications applied to other parts of the Telnet command handling.